esp/src/src-react/components/AvailablePermissions.tsx

```diff
@@ -0,0 +1,235 @@
+                }
+                return true;
+            },
+            renderHeaderCell: function (node) {
+                node.innerHTML = nlsHPCC.DenyFull;
+            }
+        }, CheckBox)
+    });
+
+    React.useCallback(() => {
+        // gridStore.on("dgrid-datachange", (evt) => {
+        //     console.log(evt);
+        // });
+    }, []);
+
+    const calcPermissionState = React.useCallback((field, value, row) => {
+        switch (field) {
+            case "allow_access":
+                row.allow_full = value && row.allow_read && row.allow_write;
+                if (value)
+                    calcPermissionState("deny_access", false, row);
+                break;
+            case "allow_read":
+                row.allow_full = row.allow_access && value && row.allow_write;
+                if (value)
+                    calcPermissionState("deny_read", false, row);
+                break;
+            case "allow_write":
+                row.allow_full = row.allow_access && row.allow_read && value;
+                if (value)
+                    calcPermissionState("deny_write", false, row);
+                break;
+            case "allow_full":
+                row.allow_access = value;
+                row.allow_read = value;
+                row.allow_write = value;
+                if (value)
+                    calcPermissionState("deny_full", false, row);
+                break;
+            case "deny_access":
+                row.deny_full = value && row.deny_read && row.deny_write;
+                if (value)
+                    calcPermissionState("allow_access", false, row);
+                break;
+            case "deny_read":
+                row.deny_full = row.deny_access && value && row.deny_write;
+                if (value)
+                    calcPermissionState("allow_read", false, row);
+                break;
+            case "deny_write":
+                row.deny_full = row.deny_access && row.deny_read && value;
+                if (value)
+                    calcPermissionState("allow_write", false, row);
+                break;
+            case "deny_full":
+                row.deny_access = value;
+                row.deny_read = value;
+                row.deny_write = value;
+                if (value)
+                    calcPermissionState("allow_full", false, row);
+                break;
+        }
+        row[field] = value;
+    }, []);
+
+    const refreshTable = React.useCallback((clearSelection = false) => {
+        grid?.set("query", gridQuery);
+        if (clearSelection) {
+            grid?.clearSelection();
+        }
+    }, [grid, gridQuery]);
+
+    const buttons = React.useMemo((): ICommandBarItemProps[] => [
+        {
+            key: "refresh", text: nlsHPCC.Refresh, iconProps: { iconName: "Refresh" },
+            onClick: () => refreshTable()
+        },
+    ], [refreshTable]);
+
+    React.useEffect(() => {
+    }, [grid, gridStore, refreshTable]);
```

Comment:
This does nothing?

Reply:
Removed the file from the PR. It was incomplete & mostly just copied directly over from the dojo component at the moment.